From e32321b4984c0dd61ace9fdea198363683c0a1a6 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Thu, 9 Jun 2016 15:09:49 +0000
Subject: [PATCH] Fix build with newer toolchain

src/MojoTriggerSubscription.cpp:63:9: error: cannot convert 'const boost::shared_ptr<MojoCall>' to 'bool' in return
src/Activity.cpp:384:9: error: cannot convert 'const boost::shared_ptr<Trigger>' to 'bool' in return
src/Activity.cpp:420:9: error: cannot convert 'const boost::shared_ptr<Callback>' to 'bool' in return
src/Activity.cpp:986:9: error: cannot convert 'const boost::shared_ptr<PowerActivity>' to 'bool' in return
---
 src/Activity.cpp                | 6 +++---
 src/MojoTriggerSubscription.cpp | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/Activity.cpp b/src/Activity.cpp
index fb0933e..9ad6ac0 100644
--- a/src/Activity.cpp
+++ b/src/Activity.cpp
@@ -381,7 +381,7 @@ void Activity::ClearTrigger()
 
 bool Activity::HasTrigger() const
 {
-	return m_trigger;
+	return m_trigger != NULL;
 }
 
 void Activity::Triggered(boost::shared_ptr<Trigger> trigger)
@@ -417,7 +417,7 @@ boost::shared_ptr<Callback> Activity::GetCallback()
 
 bool Activity::HasCallback() const
 {
-	return m_callback;
+	return m_callback != NULL;
 }
 
 void Activity::CallbackFailed(boost::shared_ptr<Callback> callback,
@@ -983,7 +983,7 @@ boost::shared_ptr<PowerActivity> Activity::GetPowerActivity()
 
 bool Activity::IsPowerActivity() const
 {
-	return m_powerActivity;
+	return m_powerActivity != NULL;
 }
 
 void Activity::PowerLockedNotification()
diff --git a/src/MojoTriggerSubscription.cpp b/src/MojoTriggerSubscription.cpp
index fada064..edccbe5 100644
--- a/src/MojoTriggerSubscription.cpp
+++ b/src/MojoTriggerSubscription.cpp
@@ -60,7 +60,7 @@ void MojoTriggerSubscription::Unsubscribe()
 
 bool MojoTriggerSubscription::IsSubscribed() const
 {
-	return m_call;
+	return m_call != NULL;
 }
 
 void MojoTriggerSubscription::ProcessResponse(MojServiceMessage *msg,
-- 
1.9.1

